src/Microsoft.AspNetCore.Dispatcher/MatcherBase.cs

```diff
@@ -14,23 +14,22 @@
 
 namespace Microsoft.AspNetCore.Dispatcher
 {
-    public abstract class DispatcherBase : IAddressCollectionProvider, IEndpointCollectionProvider
+    public abstract class MatcherBase : IMatcher, IAddressCollectionProvider, IEndpointCollectionProvider
```

Comment:
@davidfowl - do you think this base class is useful? This is something for us to discuss next meeting.